app/controllers/giving_profiles_controller.rb

```diff
@@ -0,0 +1,4 @@
+class GivingProfilesController < ApplicationController
+  def index
```

Comment:
@emilyspieler I don't think this should be an index action. 
Also, this action is supposed to be for a specific user, but right now I don't see anything in it that indicates it belongs to a specific user.

Reply:
@mintii helped me create it so she can offer input behind what the purpose of that specific controller is. Then maybe we can brainstorm best next steps.

Reply:
This controller was created for routes related to a user logged in. Totally open to your thoughts, but my interpretation was this controller would be for users logged in, which would help with managing sessions and state more closely, and another controller could be for anything not needing a user logged in and authenticated.

Reply:
We have a User model, so for the most part, user routes related routes can go into a Users controller. We can definitely have a different controller for managing the actual logging in. Maybe that can be called SessionsController or something. But creating a controller called "GivingProfilesController" doesn't really tell me anything about what it will be used for.

Reply:
I don't think we necessarily need a separate controller to represent two different states. I think the better approach is to tie like a `@current_user` to `HomeController` that will dictate which layout to show. 

That being said I'm not super opposed to have separate controllers for now to test out different layouts, but we should be more specific to the name than having it be named as our project.

Reply:
The reason I am siding with @markdamobi  on this one is that controllers are supposed to represent basically the "entry point" of our application. In our current situation, regardless of user being logged in or not will still hit the same "entry point" which in our case the `/` route. This in itself justifies the fact that we don't need two separate controllers, but instead need two separate views/layouts

Reply:
> I don't think we necessarily need a separate controller to represent two different states of an user being authenticated. I think the better approach is to tie like a `@current_user` to `HomeController` that will dictate which layout to show.
Yeah, the @current_user is what I've seen being done. Yeah, we should have the same controller regardless of the state. But it's not a bad idea to have a separate controller that actually creates the login session. After logging in, can then be redirected to some other page. I don't know about the HomeController though. I think that can create confusion since there is user home page. Is the current HomeController intended for just that?

Reply:
I already put this link on a different comment, but I'll just leave it [here again](https://trello.com/c/3Kx7QURQ/48-can-we-talk-about-specific-controller-names-and-route-names-so-were-all-on-the-same-page) as a reminder and hopefully we can talk more about it in the meeting on thursday.

Reply:
Good discussion and yes think it's worth clarifying on Thursday. Philosophically the controllers should mirror the the resources (Users, Orgs, aka the nouns) and then provide the actions one can take (view, create, update, list/index) to those resources. A login page is really just Session Create behind the scenes. The home page is an index page of all things relevant for that user.  It does get a bit squirrely at the entry point to the application. In Great Hearts we just use the home controller as a redirect based on logged in or not.

Reply:
Thanks @sanghun89 @markdamobi @uuunk and @mintii. I'll hold off on changes until we can come to a final consensus on Thursday (although it seems like we are all pretty close to being on the same page). I just want to be crystal clear on my next steps so I'll hold off on this until I can be in person with the team. Thanks for the input, everyone!